subsys/net/lib/mqtt_sn/Kconfig

```diff
@@ -27,6 +27,18 @@ config MQTT_SN_LIB_MAX_TOPIC_SIZE
 	int "Maximum topic length"
 	default 64
 
+config MQTT_SN_LIB_MAX_GATEWAYS
+	int "Maximum number of gateways to store internally"
+	default 2
```

Comment:
This should've been done for other configs, but at least for new ones it'd be good to specify `range` parameter, so that users don't set for example negative values here.

Reply:
Is there a way to define open-ended ranges in Kconfig? Or should I just make the maximum a large/unrealistic number?

Reply:
No open ranges, but we usually do something like this:
```
    range 1 $(UINT8_MAX)
```
Just choose the upper range to be something reasonable for particular config.

Reply:
Added range to all Kconfig for the mqtt_sn library